.github/workflows/test.yml

```diff
@@ -82,8 +85,8 @@ jobs:
   e2e-mariner-container:
     name: E2E testing for Mariner container
     runs-on: ubuntu-latest
+    environment: E2E
     needs: test
-    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push'
```

Comment:
We need approval for running in `E2E` environment, so we don't need this repository source check.